components/library/DateRangePicker/Preview/DateRangePicker.jsx

```diff
@@ -0,0 +1,214 @@
+        until: yesterday
+      }
+    },
+    {
+      title: 'Custom',
+      period: { since: yesterday, until: yesterday }
+    }
+  ];
+
+  const [popoverActive, setPopoverActive] = useState(false);
+  const [activeDateRange, setActiveDateRange] = useState(ranges[0]);
+  const [dateState, setDateState] = useState({
+    month: activeDateRange.period.since.getMonth(),
+    year: activeDateRange.period.since.getFullYear()
+  });
+
+  useEffect(() => {
+    setDateState({
+      month: activeDateRange.period.since.getMonth(),
+      year: activeDateRange.period.since.getFullYear()
+    });
+  }, [activeDateRange]);
+
+  const handleInputValueChange = useCallback((field, value) => {
+    if (isValidDate(value)) {
+      const newDate = new Date(value);
+      setActiveDateRange((prevState) => ({
+        ...prevState,
+        period: {
+          ...prevState.period,
+          [field]: newDate
+        }
+      }));
+    }
+  }, []);
+
+  function formatDate(date) {
+    return date.toISOString().split('T')[0];
+  }
+
+  function isValidDate(date) {
+    return !isNaN(new Date(date).getDate());
+  }
+
+  return (
+    <Box>
+      <Popover
+        active={popoverActive}
+        autofocusTarget='none'
+        preferredAlignment='left'
+        preferredPosition='below'
+        fluidContent
+        sectioned={false}
+        fullHeight
+        activator={
+          <Button icon={CalendarIcon} size='slim' onClick={() => setPopoverActive(!popoverActive)}>
+            {activeDateRange.title}
+          </Button>
+        }
+        onClose={() => setPopoverActive(false)}
+      >
+        <Box
+          style={{
+            paddingTop: '16px',
+            paddingLeft: '10px',
+            paddingRight: '10px',
+            paddingBottom: '16px'
+          }}
+        >
+          <Popover.Pane fixed>
+            <InlineGrid
+              columns={{ xs: '1fr', mdDown: '1fr', md: 'max-content max-content' }}
+              gap={0}
+            >
+              <Box
+                maxWidth='212px'
+                width='100%'
+                style={{
+                  paddingRight: '20px'
+                }}
+              >
+                <Scrollable style={{ height: '334px' }}>
+                  <OptionList
+                    options={ranges.map((range) => ({
+                      value: range.title,
+                      label: range.title
+                    }))}
+                    selected={activeDateRange.title}
+                    onChange={(selected) => {
+                      const selectedRange = ranges.find((range) => range.title === selected[0]);
+                      setActiveDateRange(selectedRange);
+                    }}
+                  />
+                </Scrollable>
+              </Box>
+              <Box maxWidth='516px'>
+                <BlockStack gap='400'>
+                  <InlineStack gap='200'>
+                    <div style={{ flexGrow: 1 }}>
+                      <TextField
```

Comment:
I'd make these text fields `readOnly` since trying to click in and type a new value doesn't really work